server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/TaskEventsSerializationTest.java

```diff
@@ -50,44 +53,50 @@
+    static final DTOConverter<TaskExecutionDetails.AdditionalInformation, AdditionalInformationDTO> ADDITIONAL_INFORMATION_CONVERTER = DTOConverter.of(MemoryReferenceWithCounterTaskAdditionalInformationDTO.SERIALIZATION_MODULE);
+    static final JsonTaskAdditionalInformationSerializer TASK_ADDITIONNAL_INFORMATION_SERIALIZER = JsonTaskAdditionalInformationSerializer.of(MemoryReferenceWithCounterTaskAdditionalInformationDTO.SERIALIZATION_MODULE);
+    static final TaskAggregateId AGGREGATE_ID = new TaskAggregateId(TaskId.fromString("2c7f4081-aa30-11e9-bf6c-2d3b9e84aafd"));
+    static final EventId EVENT_ID = EventId.fromSerialized(42);
+    static final Task TASK = new CompletedTask();
+    static final Hostname HOSTNAME = new Hostname("foo");
+    static final MemoryReferenceWithCounterTask.AdditionalInformation COUNTER_ADDITIONAL_INFORMATION = new MemoryReferenceWithCounterTask.AdditionalInformation(3, TIMESTAMP);
+
+    private final Set<EventDTOModule<?, ?>> list = TasksSerializationModule.list(
+        JsonTaskSerializer.of(
+            TestTaskDTOModules.COMPLETED_TASK_MODULE,
+            TestTaskDTOModules.MEMORY_REFERENCE_WITH_COUNTER_TASK_MODULE.apply(new MemoryReferenceWithCounterTaskStore())),
+        ADDITIONAL_INFORMATION_CONVERTER);
+
+    JsonEventSerializer serializer = JsonEventSerializer
+        .forModules(list)
+        .withNestedTypeModules(MemoryReferenceWithCounterTaskAdditionalInformationDTO.SERIALIZATION_MODULE);
 
     @ParameterizedTest
     @MethodSource
     void taskShouldBeSerializable(TaskEvent event, String serializedJson) throws Exception {
-        JsonAssertions.assertThatJson(SERIALIZER.serialize(event)).isEqualTo(serializedJson);
+        JsonAssertions.assertThatJson(serializer.serialize(event)).isEqualTo(serializedJson);
     }
 
-    private static Stream<Arguments> taskShouldBeSerializable() throws Exception {
+    static Stream<Arguments> taskShouldBeSerializable() throws Exception {
         return validTasks();
     }
 
     @ParameterizedTest
     @MethodSource
     void taskShouldBeDeserializable(TaskEvent event, String serializedJson) throws Exception {
-        assertThat(SERIALIZER.deserialize(serializedJson)).isEqualToComparingFieldByFieldRecursively(event);
+        assertThat(serializer.deserialize(serializedJson)).isEqualToComparingFieldByFieldRecursively(event);
     }
 
-    private static Stream<Arguments> taskShouldBeDeserializable() throws Exception {
+    static Stream<Arguments> taskShouldBeDeserializable() throws Exception {
         return validTasks();
     }
 
-    private static Stream<Arguments> validTasks() throws Exception {
+    static Stream<Arguments> validTasks() throws Exception {
         return Stream.of(
             Arguments.of(new Created(AGGREGATE_ID, EVENT_ID, TASK, HOSTNAME), "{\"task\":\"{\\\"type\\\":\\\"completed-task\\\"}\",\"type\":\"task-manager-created\",\"aggregate\":\"2c7f4081-aa30-11e9-bf6c-2d3b9e84aafd\",\"event\":42,\"hostname\":\"foo\"}\n"),
```

Comment:
why do we still have some embedded json in json?

Reply:
We need to do the same thing we have done for the `TaskAdditionalInformation`s to the `Task`s

Reply:
please don't make this PR bigger, create a new task instead

Reply:
if you think we can easily fix this aside of this PR, I'm ok to merge this PR in this state.